### DISCUSSION TITLE
don't understand challenge of lesson 3

### DISCUSSION THREAD

**Discussion by arbdev95** [Cyfrin/foundry-full-course-cu#146] [Q&A] [answered]
hello,

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {AFoundryCourseChallenge} from "AFoundryCourseChallenge.sol";

contract LessonOne is AFoundryCourseChallenge {
    string private constant LESSON_IMAGE = "ipfs://QmYNk9TaRp9QDRjDZYxrwh5xQGMeC9zY23tB4fs5qby6KT";

    constructor(address fcn) AFoundryCourseChallenge(fcn) {}

    /*
     * CALL THIS FUNCTION!
     * 
     * @param yourTwitterHandle - Your twitter handle. Can be a blank string.
     */
    function solveChallenge(string memory yourTwitterHandle) external {
        _updateAndRewardSolver(yourTwitterHandle);
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////////////////
    /////////////////// The following are functions needed for the NFT, feel free to ignore. ///////////////////
    ////////////////////////////////////////////////////////////////////////////////////////////////////////////
    function description() external pure override returns (string memory) {
        return "Cyfrin Foundry Full Course: Awesome! You got the basics down! You minted an NFT!";
    }

    function attribute() external pure override returns (string memory) {
        return "Getting learned!";
    }

    function specialImage() external pure override returns (string memory) {
        return LESSON_IMAGE;
    }
}

I wrote the solution(the location and newLocation uint256) in the solveChallenge, but still always gives me transaction failed, why?

**Comment by arbdev95** [comment]
if, someone can help me with this challenge,please.

**Reply by gulshanpr** [reply]
whats the problem you are getting??

**Reply by arbdev95** [reply]
when i write,after i put the uint256 of location and newLocation, and  mytwitterhandle, on metamask it says "We were not able to estimate gas. There might be an error in the contract and this transaction may fail.", can someone explain me where is the problem?

**Reply by bagariaraj23** [reply]
@arbdev95 That means that you are giving wrong values as inputs.

**Reply by cromewar** [reply, marked as answer]
Hello @arbdev95 
The position of the number changes constantly the initial position of the elements are just for the constructor, go to `read contract` and find on what position the number is, good luck!

**Reply by bagariaraj23** [reply]
just one more thing @cromewar , i have solved the challenge, but when i tried to solve it again, i gave the correct input in the location and gave a random input in newLocation. This time i got an error. Do you know anything about it

**Comment by bagariaraj23** [comment]
As far as i think,
Location : you have to enter the index value where you think that the bool value is true;
newLocation : you have to give an index as an input.

Hint: read the solveChallenge function and you'll get to know the operations and how the arrays values changes.
Hint2: Check for inherited contracts and "read" them in the etherscan carefully
